src/SerilogTracing.Instrumentation.AspNetCore/Instrumentation/AspNetCore/HttpRequestInActivityInstrumentor.cs

```diff
@@ -171,7 +171,7 @@ public void InstrumentActivity(Activity activity, string eventName, object event
 
             if (inheritTags)
             {
-                foreach (var (name, value) in incoming.EnumerateTagObjects())
+                foreach (var (name, value) in incoming.TagObjects)
```

Comment:
Defining a feature like `FEATURE_ENUMERATETAGOBJECTS` and using `#if` to switch between them here would be fine. The older API requires more allocations.

See e.g. https://github.com/serilog-tracing/serilog-tracing/blob/dev/src/SerilogTracing/SerilogTracing.csproj#L18